ICSharpCode.CodeConverter/VB/CommonConversions.cs

```diff
@@ -389,12 +397,35 @@ public static SyntaxToken ConvertIdentifier(SyntaxToken id)
             // Underscore is a special character in VB lexer which continues lines - not sure where to find the whole set of other similar tokens if any
             // Rather than a complicated contextual rename, just add an extra dash to all identifiers and hope this method is consistently used
             if (idText.All(c => c == '_')) idText += "_";
-            var keywordKind = SyntaxFacts.GetKeywordKind(idText);
-            if (keywordKind != SyntaxKind.None && !SyntaxFacts.IsPredefinedType(keywordKind))
+            
+            if (IsKeywordThatNeedsEscaped(id))
                 return SyntaxFactory.Identifier("[" + idText + "]");
             return SyntaxFactory.Identifier(idText);
         }
 
+        private static bool IsKeywordThatNeedsEscaped(SyntaxToken id)
+        {
+            var keywordKind = SyntaxFacts.GetKeywordKind(id.ValueText);
+
+            if (keywordKind == SyntaxKind.None) return false;
+            if (!SyntaxFacts.IsPredefinedType(keywordKind)) return true;
+
+            return id.IsKind(CSSyntaxKind.CatchDeclaration,
+                             CSSyntaxKind.ClassDeclaration,
+                             CSSyntaxKind.EnumDeclaration,
+                             CSSyntaxKind.EnumMemberDeclaration,
+                             CSSyntaxKind.EventDeclaration,
+                             CSSyntaxKind.EventFieldDeclaration,
+                             CSSyntaxKind.FieldDeclaration,
+                             CSSyntaxKind.InterfaceDeclaration,
+                             CSSyntaxKind.MethodDeclaration,
+                             CSSyntaxKind.PropertyDeclaration,
+                             CSSyntaxKind.NamespaceDeclaration,
+                             CSSyntaxKind.StructDeclaration,
+                             CSSyntaxKind.VariableDeclaration);
```

Comment:
Is there a link where you got this list or did you figure it out one by one? Should `DelegateDeclaration` be in here too for example?

Reply:
I'm tempted to just add all the ones ending in declaration - but then I wonder if there's a parent syntax type that might be easier to test against

```csharp
return id.IsKind(
CSSyntaxKind.VariableDeclaration,
CSSyntaxKind.CatchDeclaration,
CSSyntaxKind.NamespaceDeclaration,
CSSyntaxKind.ClassDeclaration,
CSSyntaxKind.StructDeclaration,
CSSyntaxKind.InterfaceDeclaration,
CSSyntaxKind.EnumDeclaration,
CSSyntaxKind.DelegateDeclaration,
CSSyntaxKind.EnumMemberDeclaration,
CSSyntaxKind.FieldDeclaration,
CSSyntaxKind.EventFieldDeclaration,
CSSyntaxKind.MethodDeclaration,
CSSyntaxKind.OperatorDeclaration,
CSSyntaxKind.ConversionOperatorDeclaration,
CSSyntaxKind.ConstructorDeclaration,
CSSyntaxKind.DestructorDeclaration,
CSSyntaxKind.PropertyDeclaration,
CSSyntaxKind.EventDeclaration,
CSSyntaxKind.IndexerDeclaration,
CSSyntaxKind.GetAccessorDeclaration,
CSSyntaxKind.SetAccessorDeclaration,
CSSyntaxKind.AddAccessorDeclaration,
CSSyntaxKind.RemoveAccessorDeclaration,
CSSyntaxKind.UnknownAccessorDeclaration
);
```